src/main/java/com/nextstep/lotto/domain/WinningLotto.java

```diff
@@ -6,12 +6,12 @@
     private final Lotto lotto;
     private final LottoNumber bonusNumber;
 
-    public WinningLotto(List<Integer> numbers, int bonusNumber) {
-        if (numbers.contains(bonusNumber)) {
-            throw new IllegalArgumentException("numbers cannot conatins bonus number");
+    public WinningLotto(List<LottoNumber> numbers, int bonusNumber) {
+        this.bonusNumber = LottoNumber.of(bonusNumber);
+        if (numbers.contains(this.bonusNumber)) {
+            throw new IllegalArgumentException("보너스 넘버는 당첨 번호에 포함될 수 없습니다.");
         }
         this.lotto = new Lotto(numbers);
-        this.bonusNumber = LottoNumber.of(bonusNumber);
     }
 
     public long getMatchedCount(List<LottoNumber> collectedNumbers) {
```

Comment:
```suggestion
    public long getMatchedCount(List<LottoNumber> collectedNumbers) {
        return collectedNumbers.stream()
                .filter(lottoNumber -> lotto.contains(lottoNumber))
                .count();
    }
```
get해서 리스트를 꺼내서 contains로 비교하지 말고 lotto 객체에게 물어보도록 구현해주세요
리스트 꺼내서 로직 처리하게 되면 일급컬렉션으로 만든 의미가 없어집니다

Reply:
아이고 감사합니다 😅